src/components/date-range-picker/date-range-picker.ts

```diff
@@ -0,0 +1,1211 @@
+  // Checks for dirty state to avoid unnecessary validation on form reset,
+  // caused by the inputs value being set.
+  private _delegateInputsValidity() {
+    const inputs = this.useTwoInputs ? this._inputs : [this._input];
+
+    inputs.forEach((input) => {
+      input.checkValidity = () =>
+        this._dirty || !this._pristine ? this.checkValidity() : true;
+      input.reportValidity = () =>
+        this._dirty || !this._pristine ? this.reportValidity() : true;
+    });
+  }
+
+  private _setDateConstraints() {
+    const dates: DateRangeDescriptor[] = [];
+    if (this._min) {
+      dates.push({
+        type: DateRangeType.Before,
+        dateRange: [this._min],
+      });
+      if (this._inputs[0] && this._inputs[1]) {
+        this._inputs[0].min = this._min;
+        this._inputs[1].min = this._min;
+      }
+    }
+    if (this._max) {
+      dates.push({
+        type: DateRangeType.After,
+        dateRange: [this._max],
+      });
+      if (this._inputs[0] && this._inputs[1]) {
+        this._inputs[0].max = this._max;
+        this._inputs[1].max = this._max;
+      }
+    }
+    if (!isEmpty(this.disabledDates)) {
+      dates.push(...this.disabledDates);
+    }
+
+    this._dateConstraints = isEmpty(dates) ? [] : dates;
+  }
+
+  private _updateMaskedRangeValue() {
+    if (this.useTwoInputs) {
+      return;
+    }
+    if (!(this.value?.start && this.value?.end)) {
+      this._maskedRangeValue = '';
+      return;
+    }
+    let startMask = '';
+    let endMask = '';
+    const format =
+      DateTimeUtil.predefinedToDateDisplayFormat(this._displayFormat!) ??
+      this.displayFormat ??
+      this.inputFormat;
+    if (format) {
+      startMask = DateTimeUtil.formatDate(
+        this.value.start,
+        this.locale,
+        format
+      );
+      endMask = DateTimeUtil.formatDate(this.value.end, this.locale, format);
+    } else {
+      startMask = this.value.start.toLocaleDateString();
+      endMask = this.value.end.toLocaleDateString();
+    }
+    this._maskedRangeValue = `${startMask} - ${endMask}`;
+  }
+
+  private _setCalendarRangeValues() {
+    if (!this._calendar) {
+      return;
+    }
+
+    if (!(this.value?.start || this.value?.end)) {
+      this._calendar.values = null;
+      return;
+    }
+
+    if (!(this.value?.start && this.value?.end)) {
+      const single = this.value?.start ?? this.value?.end!;
+      this._calendar.values = [single];
+      return;
+    }
+
+    const start = toCalendarDay(this.value.start);
+    const end = toCalendarDay(this.value.end);
+
+    if (start.equalTo(end)) {
+      this._calendar.values = [this.value.start!];
+    } else {
+      const range = Array.from(
+        calendarRange({ start: this.value.start!, end: this.value.end! })
+      );
+      // calendarRange is non-inclusive
+      range.push(last(range).add('day', start.lessThan(end) ? 1 : -1));
+      this._calendar.values = range.map((d) => d.native);
```

Comment:
You can extend the `calendarRange` method to support inclusive range with config parameter, right 😉 ? Better than working around it